firefox-ios/Client/Frontend/Browser/Tabs/State/TabsPanelState.swift

```diff
@@ -176,11 +196,39 @@ struct TabsPanelState: ScreenState, Equatable {
                                   isInactiveTabsExpanded: state.isInactiveTabsExpanded)
 
         default:
-            return TabsPanelState(windowUUID: state.windowUUID,
-                                  isPrivateMode: state.isPrivateMode,
-                                  tabs: state.tabs,
-                                  inactiveTabs: state.inactiveTabs,
-                                  isInactiveTabsExpanded: state.isInactiveTabsExpanded)
+            return defaultState(fromPreviousState: state)
         }
     }
+
+    // Note that the `TabsPanelState` contains a subset of all available tabs (i.e. private, normal). That means we can
+    // provide a scroll index within this tabs subarray directly as this is what the UI consumes (e.g. `TabsDisplayView`).
```

Comment:
This comment is incorrect. The TabsPanelState will only contain either private or normal tabs depending on which panel the state belongs to. See getTabsDisplayModel in TabManagerMiddleware for how the state is filtered prior to dispatching actions that pertain to TabsPanelState.

Reply:
Thanks for catching this, I forgot to update this comment last week once I realized what was actually going on!